### PR TITLE
[568560] Validation Rule and QuickFix for incoherent Instance Role names

### PR DESCRIPTION
**Pull request by sandupostaru**

### CODE REVIEW

File: core/plugins/org.polarsys.capella.core.data.interaction.validation/src/org/polarsys/capella/core/data/interaction/validation/scenario/InstanceRoleHasDifferentNameThanRepresentedInstance.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+      Scenario scenario = (Scenario) target;
+      Map<AbstractInstance, Collection<InstanceRole>> representedInstanceToInstanceRoleMap = //
+          extractRepresentedInstanceToInstanceRoleMap(scenario);
+
+      for (Map.Entry<AbstractInstance, Collection<InstanceRole>> entry : representedInstanceToInstanceRoleMap
+          .entrySet()) {
+        AbstractInstance representedInstance = entry.getKey();
+        String representedInstanceName = representedInstance.getName();
+
+        if (!StringUtil.isNullOrEmpty(representedInstanceName)) {
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.data.interaction.validation/src/org/polarsys/capella/core/data/interaction/validation/scenario/InstanceRoleHasDifferentNameThanRepresentedInstance.java:71]: if the user set to empty the represented instance, then the validation rule will not raise an error

**Review by pdulth — CHANGES_REQUESTED** [review]

File: core/plugins/org.polarsys.capella.core.data.interaction.validation/src/org/polarsys/capella/core/data/interaction/validation/scenario/InstanceRoleHasDifferentNameThanRepresentedInstance.java (hunk 1 above)
Comment (pdulth) [reply core/plugins/org.polarsys.capella.core.data.interaction.validation/src/org/polarsys/capella/core/data/interaction/validation/scenario/InstanceRoleHasDifferentNameThanRepresentedInstance.java:71]: Another validation rule will then be raised
image

**Review by pdulth — CHANGES_REQUESTED** [review]
Can you update doc to add the added validation rule.
Thanks

**Comment by sandupostaru** [comment]
> 
> 
> Can you update doc to add the added validation rule.
> Thanks

Done

**Review by pdulth — APPROVED** [review]

**Pull request merged**